infrastructure/projet/backend/main.py

```diff
@@ -0,0 +1,84 @@
+from fastapi import FastAPI, HTTPException
+from fastapi.middleware.cors import CORSMiddleware
+from virtualization_manager import create_vm_with_libvirt, create_network_with_libvirt, list_isos
+from pydantic import BaseModel
+from typing import List, Optional
+import os
+
+app = FastAPI()
+
+# Configuration CORS
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],  # Remplacez par l'URL de votre frontend
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+ISO_DIR = "/var/lib/libvirt/isos"
+
+vms_db = []
+networks_db = []
+
+class VMCreate(BaseModel):
+    name: str
+    cpu: int
+    ram: int
+    disk_size: int
+    network_name: str
+    iso: str  # Nom du fichier ISO sélectionné
+    cloud_init: Optional[dict] = None
+
+class NetworkCreate(BaseModel):
+    name: str
+    ip_address: str
+    netmask: str
+    dhcp_start: str
+    dhcp_end: str
+
+
+@app.get("/isos/", response_model=List[str])
+def get_isos():
+    print(f"{list_isos(ISO_DIR)}")
+    return list_isos(ISO_DIR)
+
+@app.post("/vms/")
+def create_vm(vm_data: VMCreate):
+    iso_path = f"{ISO_DIR}/{vm_data.iso}"
+    if not os.path.exists(iso_path):
+        raise HTTPException(status_code=404, detail=f"ISO {vm_data.iso} non trouvé")
```

Comment:
Ca ne change rien au fonctionnement mais dans FastAPI, ils ont défini toutes les constantes pour le code de retour HTTP ce qui rend le compte plus lisible :
```python
from fastapi import FastAPI, HTTPException, status
```
et
```suggestion
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"ISO {vm_data.iso} non trouvé")
```